@xen-orchestra/lite/src/components/Modal.tsx

```diff
@@ -1,9 +1,159 @@
+    }
+    instance.state.buttonList = buttonList
+    instance.state.icon = icon
+    instance.state.message = message
+    instance.state.onReject = reject
+    instance.state.onSuccess = resolve
+    instance.state.showModal = true
+    instance.state.title = title
+  })
+
+export const alert = (params: GeneralParamsModal): Promise<unknown> => {
+  const buttonList: ModalButton[] = [
+    {
+      label: <IntlMessage id='ok' />,
+      color: 'primary',
+      value: 'success',
+    },
+  ]
+  return modal({ ...params, buttonList })
+}
+
+export const confirm = (params: GeneralParamsModal): Promise<unknown> => {
+  const buttonList: ModalButton[] = [
+    {
+      label: <IntlMessage id='confirm' />,
+      value: 'success',
+      color: 'success',
+    },
+    {
+      label: <IntlMessage id='cancel' />,
+      color: 'secondary',
+      reason: 'reject',
+    },
+  ]
+  return modal({ ...params, buttonList })
 }
 
-export const confirm = ({ message }: GeneralParamsModal): Promise<string> =>
-  new Promise((resolve, reject) =>
-    window.confirm(message.props.id ?? message.props.children) ? resolve('Success') : reject(new Error())
-  )
+interface ParentState {}
+
+interface State {
+  buttonList?: ModalButton[]
+  icon?: IconName
+  message?: string | React.ReactNode
+  onReject?: (reason: unknown) => void
+  onSuccess?: (value: unknown) => void
+  showModal: boolean
+  title?: string | React.ReactNode
+}
+
+interface Props {}
+
+interface ParentEffects {}
+
+interface Effects {
+  _closeModal: () => void
+  _reject: (reason: unknown) => void
+}
+
+interface Computed {
+  buttons: ((cb: () => void) => JSX.Element)[] | undefined
+}
+
+const Modal = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: () => ({
+      buttonList: undefined,
+      icon: undefined,
+      message: undefined,
+      onReject: undefined,
+      onSuccess: undefined,
+      showModal: false,
+      title: undefined,
+    }),
+    effects: {
+      initialize: function () {
+        if (instance !== undefined) {
+          throw new Error('Modal is a singelton')
+        }
+        instance = this
+      },
+      _closeModal: function () {
+        this.state.showModal = false
+      },
+      _reject: function (reason) {
+        this.state.onReject?.(reason)
+        this.effects._closeModal()
+      },
+    },
+    computed: {
+      buttons: ({ buttonList, onSuccess, onReject }) =>
+        buttonList?.map(({ label, value, reason, ...props }, index) => {
+          const _button = (cb: () => void) => {
+            const onClick = () => {
+              if (value !== undefined) {
+                onSuccess?.(value)
+              } else {
+                onReject?.(reason)
+              }
+              cb()
+            }
+            return (
+              <Button key={index} onClick={onClick} {...props}>
+                {label}
+              </Button>
+            )
+          }
+          return _button
```

Comment:
Inline

Reply:
I cannot inline this part or eslint trigger an error: `Component definition is missing display name`

Reply:
you can use   `// eslint-disable-next-line react/display-name`.